scene/gui/line_edit.cpp

```diff
@@ -341,13 +341,15 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 				}
 
 				selection.drag_attempt = false;
-
-				if ((caret_column < selection.begin) || (caret_column > selection.end) || !selection.enabled) {
-					deselect();
-					selection.start_column = caret_column;
-					selection.creating = true;
-				} else if (selection.enabled && !selection.double_click) {
-					selection.drag_attempt = true;
+				if (!selection.double_click) {
+					bool is_inside_sel = selection.enabled && caret_column >= selection.begin && caret_column <= selection.end;
+					if (drag_and_drop_selection_enabled && is_inside_sel) {
```

Comment:
This condition is not equivalent to how it was before and makes it more difficult to drag the text.

Reply:
This change was made to make the behavior the same as how TextEdit behaves when drag-and-drop is disabled. As far as I can tell, the conditions for starting a drag and drop remain the same (aside from the new drag_and_drop_selection_enabled variable check), unless there is a case I am missing?

The intended behavior change was to create a new selection when the mouse is dragged inside of an existing selection when drag_and_drop_selection_enabled is set to false, which is how TextEdit behaves.

Reply:
Ok I tested again and dragging doesn't really feel worse. idk what happened before, I think it might have something to do with double click.